esi/lease/v1/_proxy.py

```diff
@@ -0,0 +1,245 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from esi.lease.v1 import _common
+from esi.lease.v1 import event as _event
+from esi.lease.v1 import lease as _lease
+from esi.lease.v1 import node as _node
+from esi.lease.v1 import offer as _offer
+from esi.lease.v1 import offer_claim as _offer_claim
+
+from openstack import proxy
+from openstack import resource
+
+
+class Proxy(proxy.Proxy):
+
+    _resource_registry = {
+        "offer": _offer.Offer,
+        "offer_claim": _offer_claim.OfferClaim,
```

Comment:
I'm not sure creating a new resource for claiming an offer is the right approach; for comparison, the baremetal section of the openstacksdk includes a node resource that has a large number of operations beyond CRUD. Can we parallel that approach here?

Reply:
Ah, yes! I can rewrite offer_claim similar to this: https://github.com/openstack/openstacksdk/blob/07821760224d72310c42ab4ba9a7739aa92d5eb5/openstack/baremetal/v1/node.py#L383